Jenkinsfile

```diff
@@ -1 +1,56 @@
-buildPlugin(platforms: ['linux'], jdkVersions: [7, 8])
+/* Only keep the 10 most recent builds. */
+properties([[$class: 'BuildDiscarderProperty',
+                strategy: [$class: 'LogRotator', numToKeepStr: '10']]])
+
+// TODO: Move it to Jenkins Pipeline Library
+
+/* These platforms correspond to labels in ci.jenkins.io, see:
+ *  https://github.com/jenkins-infra/documentation/blob/master/ci.adoc
+ */
+List platforms = ['linux', 'windows']
+Map branches = [:]
+
+for (int i = 0; i < platforms.size(); ++i) {
+    String label = platforms[i]
+    branches[label] = {
+        node(label) {
+            timestamps {
+                stage('Checkout') {
+                    checkout scm
+                }
+
+                stage('Build') {
+                    withEnv([
+                        "JAVA_HOME=${tool 'jdk8'}",
+                        "PATH+MVN=${tool 'mvn'}/bin",
+                        'PATH+JDK=$JAVA_HOME/bin',
+                    ]) {
+                        timeout(30) {
+                            String command = 'mvn --batch-mode clean install -Dmaven.test.failure.ignore=true'
+                            if (isUnix()) {
+                                sh command
+                            }
+                            else {
+                                bat command
+                            }
+                        }
+                    }
+                }
+
+                stage('Archive') {
+                    /* Archive the test results */
+                    junit '**/target/surefire-reports/TEST-*.xml'
```

Comment:
I don't see the test results in the archive of the CI build, is the glob pattern correct? I am looking at https://ci.jenkins.io/job/jenkinsci-libraries/job/lib-jenkins-maven-embedder/job/PR-16/.